docs/spec.md

```diff
@@ -2322,48 +2322,47 @@ planning to address this in
 
 #### Inputs
 
-| Label | Name                         | Type                                                         | Constraints                           |
-|-------|------------------------------|--------------------------------------------------------------|---------------------------------------|
-| (I1)  | `lhs`                        | tensor                                                       | (C1), (C6), (C7), (C10), (C11), (C13) |
-| (I2)  | `rhs`                        | tensor                                                       | (C1), (C8), (C9), (C10), (C11), (C13) |
-| (I3)  | `lhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64`                 | (C2), (C4), (C6), (C10), (C13)        |
-| (I4)  | `rhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64`                 | (C2), (C5), (C8), (C10)               |
-| (I5)  | `lhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64`                 | (C3), (C4), (C7), (C11)               |
-| (I6)  | `rhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64`                 | (C3), (C5), (C9), (C11)               |
-| (I7)  | `precision_config`           | variadic number of enums of `DEFAULT`, `HIGH`, and `HIGHEST` | (C12)                                 |
+| Label | Name                         | Type                                                         | Constraints                    |
+|-------|------------------------------|--------------------------------------------------------------|--------------------------------|
+| (I1)  | `lhs`                        | tensor                                                       | (C5), (C6), (C9), (C10), (C12) |
+| (I2)  | `rhs`                        | tensor                                                       | (C7), (C8), (C9), (C10), (C12) |
+| (I3)  | `lhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64`                 | (C1), (C3), (C5), (C9), (C12)  |
+| (I4)  | `rhs_batching_dimensions`    | 1-dimensional tensor constant of type `si64`                 | (C1), (C4), (C7), (C9)         |
+| (I5)  | `lhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64`                 | (C2), (C3), (C6), (C10)        |
+| (I6)  | `rhs_contracting_dimensions` | 1-dimensional tensor constant of type `si64`                 | (C2), (C4), (C8), (C10)        |
+| (I7)  | `precision_config`           | variadic number of enums of `DEFAULT`, `HIGH`, and `HIGHEST` | (C11)                          |
 
 #### Outputs
 
 | Name     | Type   | Constraints |
 |----------|--------|-------------|
-| `result` | tensor | (C13)       |
+| `result` | tensor | (C12)       |
 
 #### Constraints
 
-* (C1) `lhs` and `rhs` have the same element type.
```

Comment:
What other ops need to be updated in a similar way? E.g. probably ConvolutionOp as well at the very least?

Reply:
Thanks for pointing out. 

Other than ConvolutionOp, there are also a set of ops mentioned in #369, where the spec says that the element type has to be equal but, as per HLO semantics,  they can be off by precision. I note that the the problem about the constraints checks of dot_general to be too strict [#1439](https://github.com/openxla/stablehlo/issues/1439) is equally relevant for others ops (mentioned in #369).

I realized that mere removing the constraint will not do the job. As you pointed out, the semantics section, as it right now, does not allow lhs and rhs to be differently typed. As part of this fix,  the semantics section needs to be updated with the interpretation of different typed operands, which is something we cannot do unless we  have a clear story on precision_config #755.   

IMO, we should revisit this as part of solving precision config.  Let me know what do you think? 

PS: Will also check with the author of #1439 for feedback on the decision.